jvm/src/main/scala/org/portablescala/reflect/Reflect.scala

```diff
@@ -0,0 +1,163 @@
+package org.portablescala.reflect
+
+import scala.collection.mutable
+
+import java.lang.reflect._
+
+import org.portablescala.reflect.annotation._
+
+object Reflect {
+  /** Reflectively looks up a loadable module class using `Reflect`'s own class
+   *  loader.
+   *
+   *  A module class is the technical term referring to the class of a Scala
+   *  `object`. The object or one of its super types (classes or traits) must
+   *  be annotated with
+   *  [[org.portablescala.reflect.annotation.EnableReflectiveInstantiation @EnableReflectiveInstantiation]].
+   *  Moreover, the object must be "static", i.e., declared at the top-level of
+   *  a package or inside a static object.
+   *
+   *  If the module class cannot be found, either because it does not exist,
+   *  was not `@EnableReflectiveInstantiation` or was not static, this method
+   *  returns `None`.
+   *
+   *  This method is equivalent to calling
+   *  {{{
+   *  Reflect.lookupLoadableModuleClass(fqcn, Reflect.getClass.getClassLoader)
+   *  }}}
+   *
+   *  @param fqcn
+   *    Fully-qualified name of the module class, including its trailing `$`
+   */
+  def lookupLoadableModuleClass(fqcn: String): Option[LoadableModuleClass] =
+    lookupLoadableModuleClass(fqcn, this.getClass.getClassLoader)
```

Comment:
I think I have already mentioned this, but this method is evil: It does not properly delegate to the parent class loaders.

If a class calls `lookupLoadableModuleClass("my.dependency")` it is hostile to being loaded by a private class loader (if the `org.portablescala.reflect` package is loaded in a parent). We really should find a better solution for this :-/

Reply:
The only thing I can think of right now to solve this is a macro :(

Reply:
The macro is probably the only way, yes. It could splice in `this.getClass.getClassLoader` at call site. Hoping that blackbox macros can materialize the `this` of their call site ... (probably they can)